File: MapboxVelocity/MapboxVelocity/Core/Protocols/GeocodingServiceProtocol.swift
//
//  GeocodingServiceProtocol.swift
//  MapboxVelocity
//
//  Created by Shaikat on 4/26/26.
//
import CoreLocation

protocol GeocodingServiceProtocol: AnyObject {
    func search(query: String,
                completion: @escaping ([GeocodingResult]) -> Void)
    func reverseGeocode(coordinate: CLLocationCoordinate2D,
                        completion: @escaping (GeocodingResult?) -> Void)
}

File: MapboxVelocityAndroid/app/src/main/java/com/mapboxvelocity/Core/Protocols/GeocodingServiceProtocol.kt
package com.mapboxvelocity.Core.Protocols

import com.mapbox.geojson.Point
import com.mapboxvelocity.Core.Models.GeocodingResult

interface GeocodingServiceProtocol {
    suspend fun search(query: String): List<GeocodingResult>
    suspend fun reverseGeocode(point: Point): GeocodingResult?
}
